hub/integrations/tf/hubtensorflowdataset.py

```diff
@@ -0,0 +1,177 @@
+"""
+Tensorflow dataset wrapper
+"""
+import abc
+
+import tensorflow as tf
+
+
+def hub_tf_adapter(_fn):
+    """
+    Decorator function
+    """
+
+    def inner(*args, **kwargs):
+        return DatasetAdapter(_fn(*args, **kwargs))
+
+    return inner
+
+
+# pylint: disable=too-many-public-methods
+class HubTensorflowDataset(tf.data.Dataset):
+    """Represents a potentially large set of elements.
+
+    A `HubTensorflowDataset` can be used to represent an input pipeline as a
+    collection of elements and a "logical plan" of transformations that act on
+    those elements.
+    """
+
+    def __init__(self):
+        try:
+            variant_tensor = self._as_variant_tensor()
+        except AttributeError as attr_ex:
+            if "_as_variant_tensor" in str(attr_ex):
+                raise AttributeError(
+                    "Please use _variant_tensor instead of "
+                    "_as_variant_tensor() to obtain the variant "
+                    "associated with a dataset"
+                ) from attr_ex
+        super(__class__, self).__init__(variant_tensor)
+
+    @abc.abstractmethod
+    def _as_variant_tensor(self):
+        """Creates a scalar `tf.Tensor` of `tf.variant` representing this dataset.
+        Returns:
+          A scalar `tf.Tensor` of `tf.variant` type, which represents this dataset.
+        """
+        raise NotImplementedError("Dataset._as_variant_tensor")
+
+    @staticmethod
+    @hub_tf_adapter
+    def from_tensors(tensors):
+        return tf.data.Dataset.from_tensors(tensors)
+
+    @staticmethod
+    @hub_tf_adapter
+    def from_tensor_slices(tensors):
+        return tf.data.Dataset.from_tensor_slices(tensors)
+
+    @staticmethod
+    @hub_tf_adapter
+    def from_generator(*args, **kwargs):
+        return tf.data.Dataset.from_generator(*args, **kwargs)
+
+    @staticmethod
+    @hub_tf_adapter
+    def range(*args, **kwargs):
+        return tf.data.Dataset.range(*args, **kwargs)
+
+    @staticmethod
+    @hub_tf_adapter
+    def zip(datasets):
+        return tf.data.Dataset.zip(datasets)
```

Comment:
why is super used in some methods and tf.data.Dataset? let's use super everywhere?